chore(style-editor): Refactor style editor components for improved configuration and typing.

KevinDavilaDotCMS opened the pull request:
- Updated  to return  for default values when no options are available.
- Enhanced  to include a placeholder attribute for input fields.
- Modified  to wrap images in a container for better styling.
- Adjusted SCSS for radio components to improve layout and hover effects.
- Refined checkbox group normalization logic in  to derive default values from options.
- Updated type definitions in  to support new checkbox option structure with  and  properties.
- Improved type safety for dropdown and radio fields in  to ensure correct default value handling.

These changes enhance the flexibility and usability of the style editor, ensuring a more consistent developer experience.

### Proposed Changes
* change 1
* change 2

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

This PR fixes: #33810

This PR fixes: #33810

kevindaviladev committed:
Refactor style editor components for improved configuration and type safety

- Updated  to return  for default values when no options are available.
- Enhanced  to include a placeholder attribute for input fields.
- Modified  to wrap images in a container for better styling.
- Adjusted SCSS for radio components to improve layout and hover effects.
- Refined checkbox group normalization logic in  to derive default values from options.
- Updated type definitions in  to support new checkbox option structure with  and  properties.
- Improved type safety for dropdown and radio fields in  to ensure correct default value handling.

These changes enhance the flexibility and usability of the style editor, ensuring a more consistent developer experience.

nicobytes commented on core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-uve-palette/components/dot-uve-style-editor-form/components/uve-style-editor-field-radio/uve-style-editor-field-radio.component.scss:
```diff
@@ -35,32 +35,35 @@
     display: flex;
     flex-direction: column;
-    align-items: center;
+    align-items: start;
```
We should use PrimeFlex classes

kevindaviladev committed:
Enhance checkbox and radio field components for improved layout and styling

- Updated HTML structure for checkbox and radio components to utilize flexbox for better alignment.
- Removed redundant SCSS styles for checkbox and radio groups, simplifying the styling.
- Adjusted class names to ensure consistent application of flex properties across components.

These changes improve the visual consistency and usability of the style editor components.

kevindaviladev committed:
Removed duplicated height
